com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -289,6 +289,17 @@ public static ObservationProto GetObservationProto(this ISensor sensor, Observat
             }
             else
             {
+                // Check capabilities if we need to concatenate PNGs
+                if (sensor.GetCompressionType() == SensorCompressionType.PNG && shape.Length >= 3 &&
+                    shape[shape.Length - 1] > 3)
+                {
+                    var trainerCanHandle = Academy.Instance.TrainerCapabilities == null || Academy.Instance.TrainerCapabilities.m_ConcatenatedPngObservations;
+                    if (!trainerCanHandle)
+                    {
+                        throw new UnityAgentsException("Attached trainer doesn't support multiple PNGs. Upgrade to a version with communicator API >= 1.1.0");
```

Comment:
We could also just "downgrade" to uncompressed observations in this case (if we move the check earlier).

Reply:
Went with this instead.